lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -445,61 +406,139 @@ function Get-CharmServices {
             )
         }
     }
-
+    $s2dContextGen = @{
+        "generator" = (Get-Item "function:Get-S2DCSVContext").ScriptBlock
+        "relation" = "csv"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'csv')) {
+        $s2dContextGen['mandatory'] = $true
+    }
+    $freeRdpContextGen = @{
+        "generator" = (Get-Item "function:Get-FreeRDPContext").ScriptBlock
+        "relation" = "free-rdp"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'free-rdp')) {
+        $freeRdpContextGen['mandatory'] = $true
```

Comment:
I understand that it avoids confusions, but imagine the `s2d` relation. That would be a candidate for non-mandatory relations, since the charm can function very well without the `s2d` relation.